src/local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.cc

```diff
@@ -0,0 +1,228 @@
+#include <jni.h>
+
+#include "local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.h"
+#include "local_scheduler_client.h"
+#include "logging.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+class UniqueIdFromJByteArray {
+ private:
+  JNIEnv *_env;
+  jbyteArray _bytes;
+
+ public:
+  UniqueID *PID;
+
+  UniqueIdFromJByteArray(JNIEnv *env, jbyteArray wid) {
+    _env = env;
+    _bytes = wid;
+
+    jbyte *b =
+        reinterpret_cast<jbyte *>(_env->GetByteArrayElements(_bytes, NULL));
+    PID = reinterpret_cast<UniqueID *>(b);
+  }
+
+  ~UniqueIdFromJByteArray() {
+    _env->ReleaseByteArrayElements(_bytes, reinterpret_cast<jbyte *>(PID), 0);
+  }
+};
+
+/*
+ * Class:     org_ray_spi_impl_DefaultLocalSchedulerClient
+ * Method:    _init
+ * Signature: (Ljava/lang/String;[B[BZJ)J
+ */
+JNIEXPORT jlong JNICALL
+Java_org_ray_spi_impl_DefaultLocalSchedulerClient__1init(JNIEnv *env,
+                                                         jclass,
+                                                         jstring sockName,
+                                                         jbyteArray wid,
+                                                         jbyteArray actorId,
+                                                         jboolean isWorker,
+                                                         jlong numGpus) {
+  // 	native private static long _init(String localSchedulerSocket,
+  //     byte[] workerId, byte[] actorId, boolean isWorker, long numGpus);
+  UniqueIdFromJByteArray w(env, wid), a(env, actorId);
+  const char *nativeString = env->GetStringUTFChars(sockName, JNI_FALSE);
+  auto client = LocalSchedulerConnection_init(nativeString, *w.PID, isWorker);
+  env->ReleaseStringUTFChars(sockName, nativeString);
+  return (jlong) client;
+}
+
+/*
+ * Class:     org_ray_spi_impl_DefaultLocalSchedulerClient
+ * Method:    _submitTask
+ * Signature: (JLjava/nio/ByteBuffer;II)V
+ */
+JNIEXPORT void JNICALL
+Java_org_ray_spi_impl_DefaultLocalSchedulerClient__1submitTask(
+    JNIEnv *env,
+    jclass,
+    jlong c,
+    jbyteArray cursorId,
+    jobject buff,
+    jint pos,
+    jint sz) {
+  // task -> TaskInfo (with FlatBuffer)
+  // native private static void _submitTask(long client, /*Direct*/ByteBuffer
+  // task);
+  auto client = reinterpret_cast<LocalSchedulerConnection *>(c);
+  TaskSpec *task =
+      reinterpret_cast<char *>(env->GetDirectBufferAddress(buff)) + pos;
+  std::vector<ObjectID> execution_dependencies;
+  if (cursorId != NULL) {
```

Comment:
In general, we should avoid using `NULL` and instead use `nullptr` for pointers. However, I haven't worked much with JNI so should it be different in this case?

Reply:
I just fix them, it is pointer here